drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,373 @@
+
+    # extract dymamic capabilities
+    capabilities, _ = find_capabilities(rules, extractor)
+
+    # analysis_dir is returned for conformity reasons with the ttp generation function (same routine for both static and dynamic capability extraction)
+    return analysis_dir, capabilities
+
+
+def get_process_memory_dumps(analysis_dir: Path, pid: int) -> Iterator[str]:
+    """get the memory dumps from a specific process."""
+    # this is used mainly to get the memdumps by a malware process
+    with (analysis_dir / "memdump.log").open("r") as f:
+        for line in f:
+            dump = orjson.loads(line)
+            if dump["PID"] == pid:
+                yield dump["DumpFilename"]
+
+
+def static_capa_analysis(
+    dump_path: Path, rules: RuleSet
+) -> Tuple[Path, crd.MatchResults]:
+
+    """get the input file's capa format"""
+    try:
+        input_format = get_auto_format(dump_path)
+    except UnsupportedFormatError:
+        logger.debug("dump %s has an unsupported format", dump_path)
+        return
+
+    # get the input file's adequate analysis backend
+    if input_format == FORMAT_DOTNET:
+        backend = BACKEND_DOTNET
+    else:
+        backend = BACKEND_VIV
+
+    # get the right capa feature extractor for the input: Vivisect, .NET, etc.
+    extractor = get_extractor(
+        dump_path,
+        input_format,
+        OS_WINDOWS,
+        backend,
+        [],
+        False,
+        False,
+        dump_path,
+    )
+
+    # extract capabilities from the file, and ignore the returned metadata
+    capabilities, _ = find_capabilities(rules, extractor, disable_progress=True)
+
+    # return the capabilities alongside the memdump's path (for TTP categorization purposes)
+    return dump_path, capabilities
+
+
+def static_memory_dumps_capa_analysis(
+    analysis_dir: Path, rules: RuleSet, malware_pids: List[int] = []
+) -> Iterator[Tuple[Path, crd.MatchResults]]:
+    malware_dumps = list(
+        itertools.chain(
+            *(get_process_memory_dumps(analysis_dir, pid) for pid in malware_pids)
+        )
+    )
+
+    with TemporaryDirectory() as dump_extraction_directory:
+        # extract all memory dumps temporarily into a dumps/ folder
+        with zipfile.ZipFile(analysis_dir / "dumps.zip", "r") as zip_ref:
+            # extract the memory dumps into the temporary directory
+            zip_ref.extractall(dump_extraction_directory)
+            dumps = Path(dump_extraction_directory) / "dumps"
+
+    # extract the capabilities within each memory dump, one per thread
+    pool = multiprocessing.Pool(processes=len(4))
+    yield from pool.starmap(
+        static_capa_analysis, map(lambda dump: (dumps / dump, rules), malware_dumps)
+    )
+
+    # try to remove the temporarily created folder
+    try:
+        shutil.rmtree(dumps)
+    except PermissionError:
+        logger.debug(
+            "Permission Denied: Could not remove temporary dumps.zip extraction folder"
+        )
```

Comment:
It's good to make temporary directory cleanup in try..finally in case something will throw an exception.